### ISSUE TITLE
The action hook event is not up to date with the schema

### ISSUE THREAD

**Issue by roceroo** [directus/directus#11421]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

When I try to create a new field in a recent created collection, the schema when the hook is fired (action) is not up to date with what the schema should look like

### To Reproduce

create a action hook, try to log the collections from the context schema, create a collection in the app, the new collection will not show at the schema

### What version of Directus are you using?

9.5.0

### What version of Node.js are you using?

16.13.1

### What database are you using?

sql

### What browser are you using?

chrome

### What operating system are you using?

windows

### How are you deploying Directus?

Docker

**Comment by rijkvanzanten** [comment]
What action hook are you using?

**Comment by roceroo** [comment]
I'm using 'collections.create'
a friend of mine found this snippet that might be source of the issue (cache) 
![image](https://user-images.githubusercontent.com/47954706/152438447-dcad6d01-690c-4df7-80ab-c12e48ecd025.png)

**Comment by roceroo** [comment]
`

action('collections.create', async ({ key: collection }, { accountability }) => {
    const { FieldsService } = services
    const schema = await getSchema()

    const fieldsService = new FieldsService({
      knex: database, schema, accountability: { ...accountability, admin: true }
    })

    const fieldOptions = {
      field: 'slug',
      type: 'string',
      meta: {
        interface: 'input',
        readonly: true
      }
    }

    try {
      await fieldsService.createField(collection, fieldOptions)
    } catch (error) {
      console.log(error);
    }
  })

`

This is the code I was trying, if the schema was up to date, the new collection should show at the schema @rijkvanzanten ijkvanzanten

**Comment by roceroo** [comment]
```js
action('collections.create', async ({ key: collection }, context) => {
    const schema = await getSchema()
    console.log(schema, collection, !!schema.collections[collection]);
  })
```

if you try to run this code, you'll see that the new collection is not at the schema

**Comment by BranAndSceolan** [comment]
Hello!
I would like to try solving this issue.

**Comment by BranAndSceolan** [comment]
Sadly I had some problems getting directus to run on my pc. Since I don't have time right now to find out why, I'll unasign myself. 
If in the future there is still no solution and I got directus working on my pc I might come back to this.

**Comment by ykabusalah** [comment]
Hello, could I claim this issue and attempt fixing it?

**Comment by benhaynes** [comment]
Of course! We've assigned you, feel free to reach out over [Discord](https://directus.chat) if you have any questions along the way.

**Comment by licitdev** [comment]
To be updated after #14981 is merged. The `clearSystemCache()` needs to run before to retrieval of the updated schema. The updated schema is then passed into the `action` event.
